core/new-gui/src/app/workspace/component/workflow-editor/workflow-editor.component.ts

```diff
@@ -43,10 +44,13 @@ export class WorkflowEditorComponent implements AfterViewInit {
 
     this.createJointjsPaper();
 
+    Observable.fromEvent(window, 'resize').subscribe(
+      resizeEvent => {this.paper.setDimensions(this.getWrapperElementSize().width, this.getWrapperElementSize().height); }
+    );
+
```

Comment:
For this observable, it will fire the resizeEvent many times when user resize the browser. One suggestion I will give would be making this resizeEvent not as rapid as it is now. One observable function that you can look into is `rxjs.throttleTime`

Reply:
Based on [this](https://medium.com/@jvdheijden/rxjs-throttletime-debouncetime-and-audittime-explained-in-examples-c393178458f3) article, I feel that auditTime is better. What's your take?

Reply:
Yep, from the article's argument, seems like auditime is the better choice.

Reply:
Agreed, seems like auditime will work better in this case

Reply:
Observable.fromEvent(window, 'resize').pipe(auditTime(1000)).subscribe(
      resizeEvent => {this.paper.setDimensions(this.getWrapperElementSize().width, this.getWrapperElementSize().height); }
    );
is what I am thinking of. Is this correct? I don't know exact usage of pipe() but feel that it is used to transform observable stream

Reply:
pipe() is new in newer rxjs versions
so usually we are writing for example:
`observable.map().filter().auditTime()`
it's equivalent to 
`observable.pipe(map(), filter(), auditTime())`

Honestly I like the old syntax more. I don't think you need to use .pipe here, just try `Observable.fromEvent(window, 'resize').auditTime().subscribe()`